tests/Makefile

```diff
@@ -373,6 +377,17 @@ ifdef XILINX_SDX
 	xocc $(XOCCFLAGS) --target $(DEVICE_MODE) -l \
 	  -k TRISYCL_kernel_0 -o $@ $*.kernel.xo
 else
+# Process bitcode with SYCL passes to generate kernels
+# using -reqd-work_group_size with a nullary work-item
+# size will cause a cl runtime error, probably best to
+# leave it out for non-FPGA uses for the moment.
+%.kernel.bc: %.pre_kernel.ll
+	# To debug, think about using options like -debug -stats -print-after-all...
+	$(LLVM_BUILD_DIR)/bin/opt -load $(LLVM_BUILD_DIR)/lib/SYCL.so \
+	  -globalopt -deadargelim -SYCL-args-flattening -deadargelim \
+		-SYCL-kernel-filter -globaldce -RELGCD -inSPIRation \
+		-globaldce -o $@ $<
+
```

Comment:
I am not fan of this code duplication because it will be easy to get out of sync between FPGA & GPU...
Add instead a variable `$(XILINX_SPECIFIC_PASS)`?
And actually we discussed that on the middle term we should have `-reqd-workgroup-size-1` to trigger only on single-task kernels. But this is probably OK in the meantime...